rustfmt-core/rustfmt-bin/src/bin/main.rs

```diff
@@ -35,25 +39,247 @@ fn main() {
     std::process::exit(exit_code);
 }
 
-/// Rustfmt operations.
-enum Operation {
-    /// Format files and their child modules.
-    Format {
-        files: Vec<PathBuf>,
-        minimal_config_path: Option<String>,
-    },
-    /// Print the help message.
-    Help(HelpOp),
-    /// Print version information
-    Version,
-    /// Output default config to a file, or stdout if None
-    ConfigOutputDefault { path: Option<String> },
-    /// Output current config (as if formatting to a file) to stdout
-    ConfigOutputCurrent { path: Option<String> },
-    /// No file specified, read from stdin
-    Stdin { input: String },
+/// Format Rust code
+#[derive(Debug, StructOpt, Clone)]
+#[structopt(name = "rustfmt", version = include_str!(concat!(env!("OUT_DIR"), "/version-info.txt")))]
+struct Opt {
+    /// Run in 'check' mode.
+    ///
+    /// Exits with 0 if input is formatted correctly.
+    /// Exits with 1 and prints a diff if formatting is required.
+    #[structopt(short, long)]
+    check: bool,
+    /// Specify the format of rustfmt's output.
+    #[structopt(long)]
```

Comment:
Currently on stable the help text for `--emit` excludes the nightly-only emit modes:

```bash
        --emit [files|stdout]
                        What data to emit and how
```

and on nightly all the supported emit modes are listed:
```bash
        --emit [files|stdout|coverage|checkstyle|json]
                        What data to emit and how
```

Assuming we aren't stabilizing all the other emit modes as part of this PR, I think we could maintain that dynamic help behavior via:

```suggestion
    #[cfg_attr(nightly, structopt(long, name= "files|stdout|checkstyle|json"))]
    #[cfg_attr(not(nightly), structopt(long, name= "files|stdout"))]
```

Reply:
Good catch, thank you!